src/components/MyAccount/Settings/AccountSettingsUtils.ts

```diff
@@ -100,22 +94,23 @@ export const parsePayload = (formSubmissionBody, settingsData: Patron) => {
           },
         }
         break
-      case "homeLibrary":
-        putRequestPayload[field] = fieldValue
+      case "homeLibraryCode":
+        putRequestPayload.homeLibraryCode = fieldValue
     }
     return putRequestPayload
-  }, {} as PutRequestPayloadType)
+  }, {} as SierraPatron)
 }
 
+// Combines original patron state with the payload generated by parsePayload
+// so the account settings display can update without a data refresh.
```

Comment:
This is great, how about renaming this as `updateLocalPatronData` so it's clear that it's for the local state. The current name makes me think that it's for updating the patron through the API.

Reply:
or `updateDisplayPatronData`

Reply:
I will update! I was thinking about doing a lil refactor somewhere between 1.0 and 2.0 where there are methods on the patron to update these values, and similarly for holds/checkouts updates to avoid having to maintain the patron data in the component.